policy.tf

```diff
@@ -74,9 +74,9 @@ data "aws_iam_policy_document" "s3_bucket_with_kms_policy_document_2" {
     ]
 
     condition {
-      test     = "StringNotEquals"
-      variable = "s3:x-amz-server-side-encryption"
-      values   = ["aws:kms"]
+      test = "StringNotEquals"
+      variable = "s3:x-amz-server-side-encryption-aws-kms-key-id"
```

Comment:
Does the variable here support the alias for the KMS Key? Can you test and validate? The KMS key alias should also be added in the list of values below

Reply:
The documentation suggests that use of the alias isn't supported with this behaviour. Still worth testing, otherwise nothing to change here.